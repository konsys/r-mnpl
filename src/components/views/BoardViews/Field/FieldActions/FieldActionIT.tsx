import { IFieldRent } from "../../../../../types/types";
import React from "react";

export const FieldActionIT = (rent: IFieldRent) => {
  return (
    <>
      <div className="TableFieldcard-data-rows">
        <div>
          <div className="_title">1 полe</div>
          <div className="_value _type_x">
            <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
              <path
                d="M97.33 51.18l-14.71 32.2a6.32 6.32 0 0 1-8.38 3.12l-32.2-14.7a6.32 6.32 0 0 1-3.12-8.38l14.7-32.2A6.32 6.32 0 0 1 62 28.08L94.2 42.8a6.32 6.32 0 0 1 3.13 8.38zM64.68 35.92a6 6 0 1 0-4.99 10.92 6 6 0 0 0 5-10.92zm5.94 15.91a6 6 0 1 0-5 10.93 6 6 0 0 0 5-10.92zm5.93 15.92a6 6 0 1 0-4.99 10.93 6 6 0 0 0 5-10.93zm-42.5-3.32a7.33 7.33 0 0 0 .05 6.15l-6.54 3.05a6.32 6.32 0 0 1-8.4-3.06L4.2 38.5a6.32 6.32 0 0 1 3.05-8.4l32.09-14.96a6.32 6.32 0 0 1 8.4 3.06l3.8 8.14c-.11.18-.24.35-.33.55L34.06 64.42zm-11.12-28.4a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.89-5.08zM25.1 54.9a6 6 0 1 0 5.07 10.88A6 6 0 0 0 25.1 54.9zm19.6-29.02a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.9-5.08z"
                n-clip-rule="evenodd"
                n-fill-rule="evenodd"
              ></path>
            </svg>
            × <span className="_mult_one">{rent && rent.baseRent}</span>
          </div>
        </div>
        <div>
          <div className="_title">2 поля</div>
          <div className="_value _type_x">
            <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
              <path
                d="M97.33 51.18l-14.71 32.2a6.32 6.32 0 0 1-8.38 3.12l-32.2-14.7a6.32 6.32 0 0 1-3.12-8.38l14.7-32.2A6.32 6.32 0 0 1 62 28.08L94.2 42.8a6.32 6.32 0 0 1 3.13 8.38zM64.68 35.92a6 6 0 1 0-4.99 10.92 6 6 0 0 0 5-10.92zm5.94 15.91a6 6 0 1 0-5 10.93 6 6 0 0 0 5-10.92zm5.93 15.92a6 6 0 1 0-4.99 10.93 6 6 0 0 0 5-10.93zm-42.5-3.32a7.33 7.33 0 0 0 .05 6.15l-6.54 3.05a6.32 6.32 0 0 1-8.4-3.06L4.2 38.5a6.32 6.32 0 0 1 3.05-8.4l32.09-14.96a6.32 6.32 0 0 1 8.4 3.06l3.8 8.14c-.11.18-.24.35-.33.55L34.06 64.42zm-11.12-28.4a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.89-5.08zM25.1 54.9a6 6 0 1 0 5.07 10.88A6 6 0 0 0 25.1 54.9zm19.6-29.02a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.9-5.08z"
                n-clip-rule="evenodd"
                n-fill-rule="evenodd"
              ></path>
            </svg>
            × <span className="_mult_two">{rent && rent.oneStar}</span>
          </div>
        </div>
      </div>
    </>
  );
};
