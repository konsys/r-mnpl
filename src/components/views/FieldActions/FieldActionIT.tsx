import React from "react";

export const FieldActionIT = () => {
  return 1;
};
