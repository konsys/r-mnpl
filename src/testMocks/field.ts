import { FieldStatus, FieldType, IField, IFieldAction } from "types/types";

import { FieldActions } from "components/views/BoardViews/Field/FieldActions/FieldAction";

// import {FieldStatus} from ""

export const testField: IField = {
  fieldId: 1,
  fieldPosition: 1,
  name: "testName",
  type: FieldType.COMPANY,
  price: {
    branchPrice: 1000,
    buyoutPrice: 900,
    pledgePrice: 500,
    startPrice: 2000,
  },
  rent: {
    baseRent: 200,
    monopolyRent: 250,
    oneStar: 300,
    twoStar: 1000,
    freeStar: 1500,
    fourStar: 2000,
    bigStar: 3000,
  },
};

export const testOwnedField: IField = {
  status: {
    level: 0,
    branches: 0,
    isMonopoly: false,
    fieldId: 1,
    userId: 1,
    mortgaged: 0,
    fieldActions: [IFieldAction.MORTGAGE],
  },
  fieldId: 1,
  fieldPosition: 1,
  name: "testName",
  type: FieldType.COMPANY,
  price: {
    branchPrice: 1000,
    buyoutPrice: 900,
    pledgePrice: 500,
    startPrice: 2000,
  },
  rent: {
    baseRent: 200,
    monopolyRent: 250,
    oneStar: 300,
    twoStar: 1000,
    freeStar: 1500,
    fourStar: 2000,
    bigStar: 3000,
  },
};

export const testJailField: IField = {
  fieldId: 10,
  fieldPosition: 10,
  name: "jail",
  type: FieldType.JAIL,
  fieldCorner: 2,
  fieldLine: 1,
  isJail: true,
};
