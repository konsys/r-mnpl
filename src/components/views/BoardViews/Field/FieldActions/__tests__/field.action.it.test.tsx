import { FieldActionIT } from "../FieldActionIT";
import React from "react";
import { shallow } from "enzyme";
import { testRent } from "testMocks/field.rent";

describe("Field action IT test", () => {
  it("should render properly", () => {
    expect(shallow(<FieldActionIT {...testRent} />)).toMatchSnapshot();
  });

  it("should show rent", () => {
    const rendered = shallow(<FieldActionIT {...testRent} />);
    expect(rendered.find("._mult_one").text()).toBe(
      testRent.baseRent.toString()
    );
    expect(rendered.find("._mult_two").text()).toBe(
      testRent.oneStar.toString()
    );
  });
});
