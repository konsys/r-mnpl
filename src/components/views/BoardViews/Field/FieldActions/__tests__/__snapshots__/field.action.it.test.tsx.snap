// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field action IT test should render properly 1`] = `ShallowWrapper {}`;
